### PR TITLE
Nginx TSDB Enablement

### PR DESCRIPTION
**Pull request by agithomas**
Type of change

- Enhancement

## What does this PR do?

TSDB Enablement for Nginx

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by lalit-satapathy** [comment]
Can we remove Oracle package relevant changes from this PR? This PR is only for Nginx changes.

File: packages/nginx/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.8.0"
+  changes:
+    - description: Added changes to support TSDB Enablement.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/5057
```
Comment (muthu-mps) [inline packages/nginx/changelog.yml:6]: Can we update the PR link to point to this PR ?

File: packages/nginx/data_stream/stubstatus/fields/fields.yml [hunk 2]
```diff
@@ -3,41 +3,51 @@
...
       type: long
+      metric_type: counter
       description: |
         The total number of client requests.
     - name: current
       type: long
+      metric_type: gauge
       description: |
         The current number of client requests.
     - name: reading
```
Comment (muthu-mps) [inline packages/nginx/data_stream/stubstatus/fields/fields.yml:39]: As part of the TSDB change, Do we need to change the `avg` to `max` ? I didn't see this change for `stubstatus` visualisations.

Comment (agithomas) [reply packages/nginx/data_stream/stubstatus/fields/fields.yml:39]: Please refer here 

image

Comment (muthu-mps) [reply packages/nginx/data_stream/stubstatus/fields/fields.yml:39]: Can you please check for this visualisation - `Reading / Writing / Waiting Rates [Metrics Nginx]`

File: packages/nginx/changelog.yml (hunk 1 above)
Comment (agithomas) [reply packages/nginx/changelog.yml]: Fixed

File: packages/nginx/data_stream/stubstatus/fields/fields.yml (hunk 2 above)
Comment (agithomas) [reply packages/nginx/data_stream/stubstatus/fields/fields.yml:39]: This change is needed only when the metric_type is `counter`. metric_type` having value `counter`  does not support `avg` at present. Hence the change only for the `counter` metric_type fields only.

**Review by ritalwar — APPROVED** [review]
LGTM!

**Pull request merged**